modules/nw-cudn-cr.adoc

```diff
@@ -0,0 +1,169 @@
+* `Layer2` subnets can be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+* The `Layer2` `subnets` field is mandatory when the `ipamLifecycle` field is specified.
++
+.. Create a YAML file, such as `cluster-layer-three-udn.yaml`, to define your request for a `Layer3` topology as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: <cudn_name> # <1>
+spec:
+  namespaceSelector: # <2>
+    matchExpressions: # <3>
+    - key: kubernetes.io/metadata.name # <4>
+      operator: In # <5>
+      values: ["<example_namespace_one>, <example_namespace_two>"] # <6>
+  network: # <7>
+    topology: Layer3 # <8>
+    layer3: # <9>
+      role: Primary # <10>
+      subnets: # <11>
+        - cidr: 10.100.0.0/16
+          hostSubnet: 64
+----
+<1> Name of your `ClusterUserDefinedNetwork` custom resource.
+<2> A label query over the set of namespaces that the cluster UDN applies to. Uses the standard Kubernetes `MatchLabel` selector. Must not point to `default` or `openshift-*` namespaces.
+<3> Uses the `matchExpressions` selector type, where terms are evaluated with an _*OR*_ relationship. 
+<4> Specifies the label key to match.
+<5> Specifies the operator. Valid values include: `In`, `NotIn`, `Exists`, and `DoesNotExist`.
+<6> Because the `matchExpressions` type is used, provisions namespaces matching either `<example_namespace_one>` or `<example_namespace_two>`.
+<7> Describes the network configuration.
+<8> The `topology` field describes the network configuration; accepted values are `Layer2` and `Layer3`. Specifying a `Layer3` topology type creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+<9> This field specifies the topology configuration. Valid values are `layer2` or `layer3`.
+<10> Specifies a `Primary` or `Secondary` role. `Primary` is the only `role` specification supported in {product-version}.
+<11> For `Layer3` topology types the following specifies config details for the `subnet` field:
++
+* The `subnets` field is mandatory.
+* The type for the `subnets` field is `cidr` and `hostSubnet`:
+** `cidr` is the cluster subnet and accepts a string value.
+** `hostSubnet` specifies the nodes subnet prefix that the cluster subnet is split to.
+** For IPv6, only a `/64` length is supported for `hostSubnet`.
++
+. Apply your request by running the following command:
++
+[source,terminal]
+----
+$ oc create --validate=true -f <example_cluster_udn>.yaml
+----
++
+Where `<example_cluster_udn>.yaml` is the name of your `Layer2` or `Layer3` configuration file.
+
+. Verify that your request is successful by running the following command:
++
+[source,terminal]
+----
+$ oc get clusteruserdefinednetwork <cudn_name> --validate=true -o yaml
```

Comment:
what's the point of `--validate` while GETing the resource ?... 

I don't think it even exists:
```
oc get --validate
error: unknown flag: --validate
See 'oc get --help' for usage.
```

At least on client version `4.17.14`

Reply:
I removed it, but if I remember correctly, I couldn't get the CUdn resource without --validate=true. It was weird. at any rate, removed it.